mobile-browser-based-version/server/src/logic/federated/handlers.js

```diff
@@ -0,0 +1,547 @@
+import path from 'path';
+import fs from 'fs';
+import msgpack from 'msgpack-lite';
+import * as config from '../../../server.config.js';
+import {
+  averageWeights,
+  assignWeightsToModel,
+} from '../../helpers/tfjs_helpers.js';
+import tasks from '../../tasks/tasks.js';
+import * as tf from '@tensorflow/tfjs';
+import '@tensorflow/tfjs-node';
+
+const REQUEST_TYPES = Object.freeze({
+  CONNECT: 'connect',
+  DISCONNECT: 'disconnect',
+  SELECTION_STATUS: 'selection-status',
+  AGGREGATION_STATUS: 'aggregation-status',
+  POST_WEIGHTS: 'post-weights',
+  GET_WEIGHTS: 'get-weights',
+  POST_SAMPLES: 'post-samples',
+  GET_SAMPLES: 'get-samples',
+  GET_TASKS: 'get-tasks',
+});
+/**
+ * Fraction of client gradients required on final round of federated training
+ * to proceed to the pooling step.
+ */
+const AGGREGATION_THRESHOLD = 0.8;
```

Comment:
can you give more info also on how the total number of clients is determined and when? (it might change later if someone joins or drops late). just to make clear what the fraction is of

Reply:
the doc might be outdated here and there, this is the case here, I'll update it along with the other changes you requested!